From 230068c1cb2df569285a68d8e6e71230edbf78b6 Mon Sep 17 00:00:00 2001
From: Eelco Dolstra <eelco.dolstra@logicblox.com>
Date: Wed, 31 Oct 2012 11:53:56 +0100
Subject: [PATCH 111/126] dbus-manager: fix a fatal dbus abort in
 bus_manager_message_handler()

If ListUnitFiles fails, or an OOM occurs, then dbus_message_unref()
will be called twice on "reply", causing systemd to crash.  So remove
the call to dbus_message_unref(); it is unnecessary because of
the cleanup attribute on "reply".

[zj: modified to leave one dbus_message_unref() alone, per Colin
Walters' comment.]
(cherry picked from commit 645a9e5a2bbb06464a3fba1a3501e9d79e5bbad8)
---
 src/core/dbus-manager.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/core/dbus-manager.c b/src/core/dbus-manager.c
index 2010241..3cf3e90 100644
--- a/src/core/dbus-manager.c
+++ b/src/core/dbus-manager.c
@@ -1436,7 +1436,6 @@ static DBusHandlerResult bus_manager_message_handler(DBusConnection *connection,
                 r = unit_file_get_list(m->running_as == SYSTEMD_SYSTEM ? UNIT_FILE_SYSTEM : UNIT_FILE_USER, NULL, h);
                 if (r < 0) {
                         unit_file_list_free(h);
-                        dbus_message_unref(reply);
                         return bus_send_error_reply(connection, message, NULL, r);
                 }
 
-- 
1.8.1

